internal/generator/vector/sources.go

```diff
@@ -118,6 +120,81 @@ func ExcludeContainerPaths(namespace string) string {
 	))
 }
 
+func HttpSources(spec *logging.ClusterLogForwarderSpec, op generator.Options) []generator.Element {
+	var minTlsVersion, cipherSuites string
+	if _, ok := op[generator.ClusterTLSProfileSpec]; ok {
+		tlsProfileSpec := op[generator.ClusterTLSProfileSpec].(configv1.TLSProfileSpec)
+		minTlsVersion = tls.MinTLSVersion(tlsProfileSpec)
+		cipherSuites = strings.Join(tls.TLSCiphers(tlsProfileSpec), `,`)
+	}
+
+	el := []generator.Element{}
+	for _, input := range spec.Inputs {
+		if input.Source != nil {
+			if input.Source.HTTP != nil {
+				el = append(el, HttpSource{
+					ID:            input.Source.HTTP.Name,
+					Port:          input.Source.HTTP.Port.Port,
+					Format:        input.Source.HTTP.Format,
+					LogType:       input.Source.HTTP.LogType,
+					TlsMinVersion: minTlsVersion,
+					CipherSuites:  cipherSuites,
+				})
+			}
+		}
+	}
+	return el
+}
+
+type HttpSource struct {
+	ID            string
+	Port          int32
+	Format        string
+	LogType       string
+	TlsMinVersion string
+	CipherSuites  string
+}
+
+func (HttpSource) Name() string {
+	return "httpSource"
+}
+
+func (i HttpSource) Template() string {
+	return `
+{{define "` + i.Name() + `" -}}
+[sources.{{.ID}}]
+type = "http_server"
+address = "0.0.0.0:{{.Port}}"
+decoding.codec = "json"
```

Comment:
What if the format is plain text?

Reply:
Only audit logs, as individual JSON-encoded events, one per line, and EventList JSON arrays have been tested so far, the code doesn't look at the format field. I need more clarity wrt the logType and format combinations.

Reply:
I believe the API specifies "plan text" but given there is no support for it then we should remove that enum entry

Reply:
+1, we can re-introduce later. It would be easy to implement but  I'm not sure if anybody will actually ship us logs as plain text over HTTP. All the HTTP logging formats I've seen so far wrap the original plain text message in JSON or protobuf.

Reply:
in general, my opinion is we should drop any notion of "plain text" and only accept structured logs of a well defined format.  We are otherwise back to  3.x days before CRIO where we have to figure out a bunch of stuff and it is open ended

Reply:
+1 we need to consider carefully the likely use cases and formats we will accept, hence leaving this till GA so we can think it through. There's limited value in accepting plain text logs over HTTP since  we have no information about where they came from, unlike when we are scraping from disk. For the audit case it is enough to configure just "log-type=audit" but when extending to other log types we may need to configure more "assumed source" information and/or parse it out of known log formats (leads back to your point)